state/charm.go

```diff
@@ -47,6 +47,26 @@ func (m MacaroonCache) Get(u *charm.URL) (macaroon.Slice, error) {
 	return c.Macaroon()
 }
 
+// Channel identifies and describes completely a store channel.
+// TODO (stickupkid): Add architecture.
+type Channel struct {
+	Track  string `bson:"track,omitempty"`
+	Risk   string `bson:"risk"`
+	Branch string `bson:"branch,omitempty"`
+}
+
+// CharmOrigin holds the original source of a charm. Information about where the
+// charm was installed from (charmhub, charmstore, local) and any additional
```

Comment:
If we're modeling charm origins, for all 3 types, isn't data necessary for the store and local missing?